ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java

```diff
@@ -752,6 +752,19 @@ public boolean isMergeTransitionComplete(final BeaconState state) {
     return atState(state).miscHelpers().isMergeTransitionComplete(state);
   }
 
+  public UInt64 computeMinimumViableEpoch(final UInt64 slot) {
+    final SpecMilestone milestone = atSlot(slot).getMilestone();
+    if (milestone == SpecMilestone.PHASE0 || milestone == SpecMilestone.ALTAIR) {
+      return UInt64.ZERO;
+    }
+    // for bellatrix, we would be only allowing back to altair
+    return getForkSchedule().getForks().stream()
+        .map(Fork::getEpoch)
+        .filter(epoch -> atEpoch(epoch).getMilestone() == SpecMilestone.ALTAIR)
+        .min(UInt64::compareTo)
+        .orElse(UInt64.ZERO);
```

Comment:
would be nice to be able to do this in a way that doesnt need to be maintained for each milestone...

Reply:
Yeah this definitely doesn't look at all right to me. The `SpecMilestone` enum is ordered so you probably want to get the milestone at `slot` (I'd just use `fork(computeEpochAtSlot(slot))` and then use `SpecMilestone.getAllPriorMilestones` and take the last one (could actually simplify by adding a `Optional<SpecMilestone> SpecMilestone.getPreviousMilestone()` method which could do it based on the enum ordinal.

Then just look up the epoch that previous milestone is scheduled and that's the earliest epoch you can support.